src/org/jitsi/impl/neomedia/codec/video/vp9/DePacketizer.java

```diff
@@ -166,5 +169,49 @@ public static int getSpatialLayerIndex(byte[] buf, int off, int len)
 
             return (buf[loff] & 0xE) >> 1;
         }
+        
+        /**
+         * Check if the current the packet contains a key frame
+         *
+         * @param buf the byte buffer that holds the VP9 payload.
+         * @param off the offset in the byte buffer where the VP9 payload starts.
+         * @param len the length of the VP9 payload.
+         * @return true if the frame is a key frame, false otherwise
+         */
+        public static boolean isKeyFrame(byte[] buf, int off, int len)
+        {
+            //This packet will have its P bit equal to zero, SID or D
+            //bit (described below) equal to zero, and B bit (described below)
+            //equal to 1
+            
+            if (!isValid(buf, off, len))
+            {
+                return false;
+            }
+
+            // P_BIT must be 0 and B_BIT 1
+            if((buf[off] & P_BIT) != 0 || (buf[off] & B_BIT) == 0)
+            {
+                return false;
+            }
+
+            int loff = off + 1;
+            if ((buf[off] & I_BIT) != 0)
+            {
+                loff += 1;
+                if ((buf[off + 1] & (1 << 7)) != 0)
+                {
+                    // extended pid.
+                    loff += 1;
+                }
+            }
+            //SID or D bit equal to zero
+            return ((buf[loff] & 0xE) >> 1) == 0 || (buf[loff] & 0x1) == 0;
```

Comment:
Could you please extract the mask constants to something that reads XXX_MASK?

Reply:
I think this needs to be refactored afterwards, as the same kind of refactor is needed in VP8.depacketizer. 
Having said that, do you mean SID_MASK and D_MASK?